src/pkg/repository/repository_test.go

```diff
@@ -140,6 +142,23 @@ func (suite *RepositoryIntegrationSuite) TestInitialize() {
 	}
 }
 
+func (suite *RepositoryIntegrationSuite) TestInitializeCustomCredentials() {
```

Comment:
Should we add a unit test that shows the Ark team how to use the custom credentials? I assume we need to use `NewStorageWithCredentials`.

Reply:
@grsubramanian - that's correct. I'll follow up with the additional unit test later (in `storage_test.go`) so that this can merge.